refactor and simplify mkFilter, use more restrictive rules

jakubgs opened the pull request:
This should improve #10234 which was related to the derivation for `status-react-patched-npm-gradle-modules` using too broad filter and including some files which would cause unnecessary rebuilds when running targets like `make run-android`.

The thing is, XML files like `AndroidManifest.xml` need to be part of the `status-react-source-gradle-install` derivation because they are used in the build of `status-react-patched-npm-gradle-modules`.

I also changed the `TARGET` to `default` for `make run-android` to stop rebuilds of unnecessary things.

Status: __Ready__

jakubgs committed:
refactor and simplify mkFilter, use more restrictive rules

This should reduce rebuilds of status-react-npm-gradle-modules

Signed-off-by: Jakub Sokołowski <jakub@status.im>

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 2b9e0d89 | [#1](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-10246/1/) | 2020-03-30 18:22:29 | ~4 min | `android-e2e` | [:page_facing_up:`log`](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-10246/1/consoleText) |

jakubgs committed:
use default target for run-android

this should stop rebuilds of things like
- status-react-patched-npm-gradle-modules
- status-react-source-gradle-install

Signed-off-by: Jakub Sokołowski <jakub@status.im>

flexsurfer commented on Makefile:
```diff
@@ -269,7 +269,7 @@ _run-%:
 
 # TODO: Migrate this to a Nix recipe, much the same way as nix/mobile/android/targets/release-android.nix
-run-android: export TARGET := android
+run-android: export TARGET := default
```
could you pls elaborate on how this change works?

jakubgs replied:
Sure, It's pretty simple.
There's a `nix/shells.nix` file that defines shells we use in CI and `Makefile`:
https://github.com/status-im/status-react/blob/afa2a92045ff171e7065864b64b13a693262e500/nix/shells.nix#L51-L71
These shells are defined using the `mkShell` derivation, here's a simple example:
https://github.com/status-im/status-react/blob/afa2a92045ff171e7065864b64b13a693262e500/nix/shells.nix#L25-L28
Which creates a the `keytool-sh` derivation that is then exported as `keytool` which includes `openjdk8` in `PATH` when created using `make shell TARGET=keytool`:
```
 > make shell TARGET=keytool
Configuring default Nix shell for target 'keytool'...

[nix-shell:~/work/status-react]$ which keytool
/nix/store/88zs3fkbwq5vxrynmc54j7s03vbyfbxw-openjdk-8u222-ga/bin/keytool
```
The `default` target shell is just one that includes all the tools we usually use:
https://github.com/status-im/status-react/blob/afa2a92045ff171e7065864b64b13a693262e500/nix/shells.nix#L30-L43
